lib/protoparser/datadog/stream/streamparser.go

```diff
@@ -144,15 +182,15 @@ func putPushCtx(ctx *pushCtx) {
 var pushCtxPool sync.Pool
 var pushCtxPoolCh = make(chan *pushCtx, cgroup.AvailableCPUs())
 
-func getRequest() *datadog.Request {
+func getRequest() datadog.Request {
```

Comment:
This function may return incorrect data type. 

 E.g. struct for v1 series was stored at pool and at the next call for v2 API it would be returned from pool causing parsing error.


 I suggest to create sync.Pool at v1/api, v2/api, sketeches/ packages and use GetRequest() method per API.

 Something like:
```go
var request datadog.Request

switch apiKind{
  case "v1":
     // returns *apiSeriesV1.Request
     // exact type, not interface
     request = apiSeriesV1.GetRequest()
     defer apiSeriesV1.PutRequest(request)
  case "v2":
      request = apiSeriesV2.GetRequest()
      defer apiSeriesV2.PutRequest(request)
}
```

Reply:
See pools separated here https://github.com/VictoriaMetrics/VictoriaMetrics/commit/98d0f81f21342296b20e13e2855dec9156c8b73c